src/adapter/src/catalog.rs

```diff
@@ -2445,6 +2461,25 @@ impl<S: Append> Catalog<S> {
         for (name, value) in system_config {
             catalog.state.insert_system_configuration(&name, &value)?;
         }
+        if !catalog.state.system_config().config_has_synced_once() {
```

Comment:
Let's make this way spammier! An info log at the start and end of config syncing would be great, and also an info log if we _don't_ need to do the initial config sync. Otherwise if LD goes down we're going to be mighty confused as to why some envd somewhere is wedged on boot, because we won't remember this code exists in the heat of that moment!

I think a debug log for each discovered system parameter would also be a good idea.

Reply:
Done. I used only `--log-filter mz_adapter::catalog=debug,mz_adapter::config=debug` to see only the relevant output when  testing.